lms/djangoapps/etc_api/views.py

```diff
@@ -0,0 +1,132 @@
+
+
+class CreateUserAccountWithoutPasswordView(APIView):
+    authentication_classes = (OAuth2AuthenticationAllowInactiveUser,)
+    permission_classes = ApiKeyHeaderPermission,
+
+    def post(self, request):
+        """
+        Create user account without password
+
+        Creates a user using mail, login and also name and surname.
+        Sets a random password and sends a user a message to change it.
+        """
+        data = request.data
+        data['honor_code'] = "True"
+        data['terms_of_service'] = "True"
+        data['name'] = request.data.get('prename')
+        email = request.data.get('email')
+        username = request.data.get('username')
+        try:
+            validate_slug(username)
+        except ValidationError as err:
+            return Response(
+                data={
+                    "user_message": "Enter a valid 'username' consisting of letters, numbers, underscores or hyphens."
+                },
+                status=400
+            )
+
+        if check_account_exists(username=username, email=email):
+            return Response(data={"user_message": "User already exists"}, status=409)
+
+        try:
+            data['password'] = uuid4().hex
+            user = create_account_with_params(request, data)
+            user.is_active = True
+            user.first_name = request.data.get('prename')
+            user.last_name = request.data.get('surname')
+            user.save()
+            self.send_activation_email(request)
+        except ValidationError as err:
+            # Should only get non-field errors from this function
+            assert NON_FIELD_ERRORS not in err.message_dict
+            # Only return first error for each field
+            return Response(data={"user_message": "Wrong parameters on user creation"}, status=400)
+        except ValueError as err:
+            return Response(data={"user_message": "Wrong email format"}, status=400)
+
+        return Response(data={'user_id': user.id}, status=200)
+
+    @staticmethod
+    def send_activation_email(request):
+        form = PasswordResetFormNoActive(request.data)
+        if form.is_valid():
+            form.save(use_https=request.is_secure(),
+                      from_email=configuration_helpers.get_value(
+                          'email_from_address', settings.DEFAULT_FROM_EMAIL),
+                      request=request,
+                      subject_template_name='etc_api/set_password_subject.txt',
+                      email_template_name='etc_api/set_password_email.html'
+                      )
+
+
+class SetActivateUserStatus(APIView):
+    authentication_classes = (OAuth2AuthenticationAllowInactiveUser,)
+    permission_classes = ApiKeyHeaderPermission,
+
+    def post(self, request):
+        """
+        Enable or disable user by user id.
+        """
+        data = request.data
+        try:
+            user = User.objects.get(id=data['user_id'])
```

Comment:
What if required parameters are absent in the request?

Reply:
I added code for checking user_id.